bagua/torch_api/bucket.py

```diff
@@ -203,71 +204,129 @@ def append_centralized_synchronous_op(
 
     def append_decentralized_synchronous_op(
         self,
+        peer_weight: BaguaTensor,
         hierarchical: bool = True,
         peer_selection_mode: str = "all",
-        communication_interval: int = 1,
-        compression: Optional[str] = None,
-        weight: Optional[BaguaTensor] = None,
-        left_peer_weight: Optional[BaguaTensor] = None,
-        right_peer_weight: Optional[BaguaTensor] = None,
     ) -> BaguaBucket:
         """
         Append a decentralized synchronous operation to a bucket. It will do gossipy style model averaging among workers.
 
-        The operations will be executed by the Bagua backend in the order they are appended
-        when all the tensors within the bucket are marked ready.
+        In this operation, local model weights are copied into `peer_weight`, and each worker sends `peer_weight` to its
+        peer and receives its peer's counterpart, then average them. Since the original local model weights remains
+        unchanged during this process, it is possible to overlap communication with the computation of `forward` and
+        `backward` process.
+
+        This function initiates the above mentioned process, which will be executed by the Bagua backend in
+        the order they are appended when all the tensors within the bucket are marked ready.
+
+        To complete this operation, it is needed to copy `peer_weight` back to local model weights once `backward` finishes.
+        This is done by another function, see :func:`decentralized_synchronous_op_copy_back_peer_weight`.
```

Comment:
In this operation, local model weights are copied into `self.peer_weight`, and the result of decentralized averaging will be in `self.peer_weight`.
        To copy `self.peer_weight` back to self, call `self.decentralized_synchronous_op_copy_back_peer_weight()`.

Reply:
> ```
>     This operation is not inplace, which means the bucket content is first copied to `self.peer_weight`, and the result of decentralized averaging will be in `self.peer_weight`.
>     To copy `self.peer_weight` back to self, call `self.decentralized_synchronous_op_copy_back_peer_weight()`.
> ```
It we implement it as described above, `peer_weight` in the argument list can be removed

Reply:
Oh we don't want to do hardcode self.peer_weight

```
This operation is not inplace, which means the bucket content is first copied to `peer_weight`, and the result of decentralized averaging will be in `peer_weight`. To copy `peer_weight` back to self, call `bucket.decentralized_synchronous_op_copy_back_peer_weight(...)`.
```